### ISSUE TITLE
Allow users to specify the configuration file as a command line argument (Trac #162)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#164]
By default, the Microdrop configuration file location is fixed. By allowing it to be specified as a command line argument, it would be possible to install multiple versions of Microdrop on the same computer (each with it's own config file and plugin/device directory).

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/162
```json
{
    "status": "closed", 
    "changetime": "2015-01-13T08:45:21", 
    "description": "By default, the Microdrop configuration file location is fixed. By allowing it to be specified as a command line argument, it would be possible to install multiple versions of Microdrop on the same computer (each with it's own config file and plugin/device directory).", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1421138721587479", 
    "component": "microdrop", 
    "summary": "Allow users to specify the configuration file as a command line argument", 
    "priority": "major", 
    "keywords": "", 
    "version": "", 
    "time": "2014-11-06T17:38:10", 
    "milestone": "Microdrop 1.0", 
    "owner": "", 
    "type": "enhancement"
}
```

**Comment by cfobel** [comment]
Trac update at 20150105T18:20:35: cfobel changed milestone from "Microdrop 1.1" to "Microdrop 1.0"

**Comment by cfobel** [comment]
Trac update at 20150106T18:35:03: 
* cfobel changed _comment0 from:

> {{{
> #!md
> As of 
> 
> , it is possible to specify configuration file when launching MicroDrop using the `-c` command-line flag.  Note that the `data_dir`, `plugins.directory` and `microdrop.gui.dmf_device_controller.device_directory` config settings can be set using relative paths.  By using relative paths, it is possible to make a relocatable MicroDrop configuration directory, that can be placed in whatever directory MicroDrop is launched from.  For example, consider launching MicroDrop where a data directory is copied to `.microdrop`:
> 
> ```
> python -m microdrop -c .microdrop/microdrop.ini
> ```
> 
> The following `microdrop.ini` would make things run from the local `.microdrop` data directory *(e.g., in the root of a portable Python distribution where the `microdrop` package is installed)*.
> 
> ```
> # `.microdrop/microdrop.ini`
> data_dir = .microdrop
> ...
> [plugins]
> ...
> directory = .microdrop/plugins
> ...
> [microdrop.gui.dmf_device_controller]
> ...
> device_directory = .microdrop/devices
> ...
> ```
> }}}

to:

> 1420569487919421


* cfobel commented:

> As of commit 4517c39e90707dd25fc94a1b5b9cd27edb229239, it is possible to specify configuration file when launching !MicroDrop using the `-c` command-line flag.  Note that the `data_dir`, `plugins.directory` and `microdrop.gui.dmf_device_controller.device_directory` config settings can be set using relative paths.
> 
> ```
> #!md
>   By using relative paths, it is possible to make a relocatable MicroDrop configuration directory, that can be placed in whatever directory MicroDrop is launched from.  For example, consider launching MicroDrop where a data directory is copied to `.microdrop`:
> 
> ```
> python -m microdrop -c .microdrop/microdrop.ini
> ```
> 
> The following `microdrop.ini` would make things run from the local `.microdrop` data directory *(e.g., in the root of a portable Python distribution where the `microdrop` package is installed)*.
> 
> ```
> # `.microdrop/microdrop.ini`
> data_dir = .microdrop
> ...
> [plugins]
> ...
> directory = .microdrop/plugins
> ...
> [microdrop.gui.dmf_device_controller]
> ...
> device_directory = .microdrop/devices
> ...
> ```
> ```

**Comment by cfobel** [comment]
Trac update at 20150113T08:45:21: 
* ryan commented:

> Fixed in commit  changeset f494a77.


* ryan changed resolution from "" to "fixed"
* ryan changed status from "new" to "closed"
